### ISSUE TITLE
Can't load tests of a Stack project if they don't compile

### ISSUE THREAD

**Issue by neongreen** [haskell/haskell-mode#1411]
1. Create a new Stack project: `stack new foo && cd foo`.
2. Replace `test/Spec.hs` with something that wouldn't compile, for instance

   ``` haskell
   main = foo
   ```
3. Open `src/Lib.hs` in Emacs. Load it with `C-c C-l`. It will load successfully.
4. Okay, now we want to work with tests, so do `M-x haskell-session-change-target` and choose `foo:test:foo-test`. Confirm restarting of Haskell process.
5. You'll get a message that ``The Haskell process `foo' has died. Restart?``. Pressing `l` will reveal this process log (from which I have removed `^H`s and `Progress: 0/2` messages to make it more readable):

   ```
("Starting inferior stack GHCi process using stack" "foo" nil "stack" "ghci" "foo:test:foo-test" "--ghc-options=-ferror-spans")
-> Prelude.putStrLn ""
   :set -v1
   :set +c
-> :set prompt "\4"
-> :set prompt2 "| "
<- foo-0.1.0.0: unregistering (components added: test:foo-test)
   
<- Progress: 0/2
   foo-0.1.0.0: build (lib + exe + test)
<- Preprocessing library foo-0.1.0.0...
<- In-place registering foo-0.1.0.0...
   Preprocessing executable 'foo-exe' for foo-0.1.0.0...
<- Preprocessing test suite 'foo-test' for foo-0.1.0.0...
<- [1 of 1] Compiling Main             ( test/Spec.hs, .stack-work/dist/x86_64-linux/Cabal-1.22.5.0/build/foo-test/foo-test-tmp/Main.o )
<- test/Spec.hs:1:8-10: Not in scope: ‘foo’
   
   Progress: 1/2
   --  While building package foo-0.1.0.0 using:
         /home/yom/.stack/setup-exe-cache/x86_64-linux/setup-Simple-Cabal-1.22.5.0-ghc-7.10.3 --builddir=.stack-work/dist/x86_64-linux/Cabal-1.22.5.0 build lib:foo exe:foo-exe test:foo-test --ghc-options " -ddump-hi -ddump-to-file"
       Process exited with code: ExitFailure 1
   
   Warning: build failed, but optimistically launching GHCi anyway
<- The following GHC options are incompatible with GHCi and have not been passed to it: -threaded
   Using main module: 1. Package `foo' component test:foo-test with main-is file: /tmp/foo/test/Spec.hs
   Configuring GHCi with the following packages: foo
   GHCi, version 7.10.3: http://www.haskell.org/ghc/  :? for help
   
<- <command line>: cannot satisfy -package foo-0.1.0.0
       (use -v for more information)
   
Event: "exited abnormally with code 1
"
```

If tests compile successfully and then I modify them and reload the file, the errors are processed as usual (i.e. printed into the interactive buffer).

**Comment by gracjan** [comment]
Looks like `stack` itself should handle this situation better than it does right now. Reporting upstream.

**Comment by gracjan** [comment]
Does not seem like haskell-mode can do much here. Waiting on stack to fix. Closing!
